src/components/EditImageModal/index.jsx

```diff
@@ -68,19 +69,22 @@ export default class EditImageModal extends React.Component {
         aspectRatio: img.naturalWidth / img.naturalHeight,
       },
       imageLoading: false,
+      isImageValid: true,
     });
   }
 
   onImageError = () => {
     this.setState({
+      imageDimensions: {},
       imageLoading: false,
+      isImageValid: false,
     });
   }
 
   onImageSourceBlur = (imageSource) => {
     this.setState({
       imageSource,
-      imageLoading: imageSource.length > 0,
+      imageLoading: imageSource.length > 0 && (this.state.imageSource !== imageSource),
```

Comment:
I'm wondering whether we should have more states for this image? I'm finding that I'm going to need some notion of "imageIsEmpty" and "imageIsLoaded" so that I know when to conditionally hide the placeholder div once the image is loaded. Would me adding that in my pull request conflict with anything you've done here?

Reply:
I think as long as `imageLoading` is still set when the image is loading then everything here should still work.